src/workerd/api/basics.c++

```diff
@@ -666,6 +709,97 @@ void AbortSignal::triggerAbort(
   dispatchEventImpl(js, js.alloc<Event>(kj::str("abort")));
 }
 
+void AbortSignal::serialize(jsg::Lock& js, jsg::Serializer& serializer) {
+  auto& handler = JSG_REQUIRE_NONNULL(serializer.getExternalHandler(), DOMDataCloneError,
+      "AbortSignal can only be serialized for RPC.");
+  auto externalHandler = dynamic_cast<RpcSerializerExternalHander*>(&handler);
+  JSG_REQUIRE(
+      externalHandler != nullptr, DOMDataCloneError, "AbortSignal can only be serialized for RPC.");
+
+  serializer.writeRawUint32(static_cast<uint>(canceler->isCanceled()));
+  serializer.writeRawUint32(static_cast<uint>(flag));
+  KJ_IF_SOME(r, reason) {
+    serializer.write(js, r.getHandle(js));
+  } else {
+    serializer.write(js, js.undefined());
+  }
+
+  auto& ioContext = IoContext::current();
+
+  auto streamCap =
+      externalHandler->writeStream([&](rpc::JsValue::External::Builder builder) mutable {
+    auto field = builder.initAbortSignal();
+    field.setSignal(kj::heap<AbortSignalImpl>(ioContext, JSG_THIS));
+  });
+
+  // Keep track of every AbortSignal cloned from this one.
+  // If this->triggerAbort(...) is called, each rpcClient will be informed.
+  rpcClients.add(kj::mv(streamCap).castAs<rpc::AbortSignal>());
```

Comment:
One thing that my `MessageChannel` PR did not cover yet was removing `rpcClients` from this collection when the capability is dropped remotely. You might need to make sure this doesn't become a memory leak or problematic.

Reply:
What method do I need to implement to get notified about this situation?

Reply:
heh, I hadn't quite worked that out yet for the MessageChannel PR. Still fairly new to this mechanism myself. @kentonv likely can say.

Reply:
Sounds to me like you need a membrane

Reply:
I've seen that word before. Do you remember a good file for me to look at?

Reply:
I actually don't think you need to do anything here.

These are outgoing capabilities, that is, this object is holding pointers to remote objects. They don't get "dropped remotely", because this end is the end that can be "dropped". Arguably you could try to detect if the connection has been lost, but I don't think there's any need to do that. If the capabilities are disconnected, then calling them will fail, but you can just ignore that. I don't think someone is going to send the _same_ AbortSignal over a huge number of connections in order to cause a memory leak.